toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -132,20 +132,6 @@ class DiagnosticConsumer {
   virtual auto Flush() -> void {}
 };
 
-// An interface that can translate some representation of a location into a
-// diagnostic location.
-//
-// TODO: Revisit this once the diagnostics machinery is more complete and see
-// if we can turn it into a `std::function`.
```

Comment:
I don't think we're there yet. In particular, we don't support highlights or annotations on source ranges, which are the kinds of features that I was concerned may lead to us wanting additional functions on this interface.

Reply:
Okay, I'll revert this change.